### PR TITLE
Prevent deleted API keys from being returned

### PR DESCRIPTION
**Pull request by matthewelwell**
## Changes

Ensures that deleted MasterAPIKeys are not returned by the viewset after being deleted. 

## How did you test this code?

Added a new integration test.

### CODE REVIEW

File: api/tests/integration/api_keys/test_viewset.py [hunk 1]
```diff
@@ -125,3 +125,42 @@ def test_create_master_api_key_ignores_organisation_in_body(admin_client, organi
 
     assert list_response_json["results"][0]["name"] == name
     assert key.startswith(list_response_json["results"][0]["prefix"])
+
+
+def test_deleted_api_key_is_not_returned_in_list_and_cannot_be_used(
+    admin_client, organisation, master_api_key, master_api_key_client
```
Comment (khvn26) [inline api/tests/integration/api_keys/test_viewset.py]: nit: typing

**Review by khvn26 — APPROVED** [review]

File: api/tests/integration/api_keys/test_viewset.py (hunk 1 above)
Comment (matthewelwell) [reply api/tests/integration/api_keys/test_viewset.py]: Good point - have added.

File: api/tests/integration/api_keys/test_viewset.py [hunk 2]
```diff
@@ -125,3 +126,45 @@ def test_create_master_api_key_ignores_organisation_in_body(admin_client, organi
 
     assert list_response_json["results"][0]["name"] == name
     assert key.startswith(list_response_json["results"][0]["prefix"])
+
+
+def test_deleted_api_key_is_not_returned_in_list_and_cannot_be_used(
+    admin_client: APIClient,
+    organisation: int,
+    master_api_key: dict,
```
Comment (khvn26) [inline api/tests/integration/api_keys/test_viewset.py]:
```suggestion
    master_api_key: typing.Tuple["AbstractAPIKey", str],
```

File: api/tests/integration/api_keys/test_viewset.py [hunk 3]
```diff
@@ -125,3 +126,45 @@ def test_create_master_api_key_ignores_organisation_in_body(admin_client, organi
...
+    master_api_key_client: APIClient,
+):
```
Comment (khvn26) [inline api/tests/integration/api_keys/test_viewset.py]:
```suggestion
) -> None:
```

**Review by khvn26 — CHANGES_REQUESTED** [review]

File: api/tests/integration/api_keys/test_viewset.py (hunk 3 above)
Comment (matthewelwell) [reply api/tests/integration/api_keys/test_viewset.py]: I'm not sure I agree that we should add the return type to all of our tests? I think that `None` is assumed to be the default return type, and we can be fairly confident that all test functions / methods have no return type. Happy to discuss this further though if you have a strong opinion on this.

Comment (khvn26) [reply api/tests/integration/api_keys/test_viewset.py]: I would agree with you if omitting a return type would mean an implicit `None`. Unfortunately, it means implicit `Any` , so even obvious cases like tests and `__init__` methods have to be annotated if we want to come to a fully typed project in the future.

**Review by khvn26 — APPROVED** [review]
